README.adoc

```diff
@@ -0,0 +1,223 @@
+to write code that can be maintained both, upstream and downstream.
+
+You can generate a PDF copy of this guide using https://asciidoctor.org/docs/asciidoctor-pdf/[AsciiDoctor PDF],
+and an HTML copy https://asciidoctor.org/docs/convert-documents/#converting-a-document-to-html[with] https://asciidoctor.org/#installation[AsciiDoctor]
+using the following commands:
+
+[source,shell]
+----
+# Generates README.pdf
+asciidoctor-pdf -a allow-uri-read README.adoc
+
+# Generates README.html
+asciidoctor README.adoc
+----
+
+[TIP]
+====
+Install the `rouge` gem to get nice syntax highlighting in the generated document.
+
+[source,shell]
+----
+gem install rouge
+----
+====
+
+[TIP]
+====
+https://github.com/rubocop-hq/rubocop[RuboCop], a static code analyzer (linter) and formatter,
+has a https://github.com/utkarsh2102/rubocop-packaging[`rubocop-packaging`] extension, based
+on this style guide.
+
+[source,shell]
+----
+gem install rubocop-packaging
+----
+====
+
+
+== How To Read This Guide
+
+The guide is separated into sections based on the different cops that the Packaging extension
+provides. +
+There was an attempt to explicitly mention everything so if anything is still unclear, feel
+free to open an issue asking for further clarity.
+
+
+== A Living Document
+
+This guide is a work in progress - existing guidelines are constantly being improved, new
+guidelines are being added, and occasionally some guidelines would get removed.
+
+
+== Why Packaging Extension?
+
+The Debian Ruby team has a lot of experience in packaging and maintaining Ruby libraries and
+applications for Debian. During this work, they identified several issues in upstream codebases
+that makes it difficult to build a Debian package straight out of those Ruby gems (shipped via
+https://rubygems.org[rubygems]).
+
+The Debian developers (downstream maintainers) have been in touch with the RubyGems and other
+upstream maintainers and we're collaborating to try to make things easier for OS packagers
+while not compromising the experience for upstream maintainers.
+
+As a result, we're working on this RuboCop extension to enforce a set of best practices that
+upstream maintainers can follow to make lifes of packagers easier. +
+And that is how `rubocop-packaging` is born!
+
+
+== Cops Description
+
+Listed below are the cops provided by the `Packaging` extension, describing the need to write
+and enforce it and how is it problematic on the Debian (downstream) side.
+
+=== Gemspec Git [[gemspec-git]]
+
+About the `GemspecGit` cop:
+
+==== Rationale [[gemspec-git-rationale]]
+
+Avoid using `git ls-files` to produce lists of files. Downstreams often need to build your
+package in an environment that does not have git (on purpose). Instead, use some
+pure Ruby alternative, like `Dir` or `Dir.glob`.
+
+==== Need [[gemspec-git-need]]
+
+Packages in Debian are built in a clean environment (https://wiki.debian.org/sbuild[sbuild],
```

Comment:
I think you can also mention the source package consists of released tar ball (usually downloaded from github tags/releases page or converted from the .gem file), which is extracted during build. So even if we add git as a build dependency, it would still fails as the debian package source tree is not a git repo. Even when the package is maintained in git, it is uploaded as tarballs to the archive without any version control information.

Reply:
This is a fantastic point! :heart: 
Adding.. \o/